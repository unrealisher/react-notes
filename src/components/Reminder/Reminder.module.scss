.reminder {
  padding: 12px 24px;
  padding-bottom: 28px;
  padding-left: 60px;
  background-color: #db5341;
  margin-bottom: -16px;
}

.text {
  position: relative;
  color: white;

  &::before {
    content: "";
    position: absolute;
    width: 20px;
    height: 20px;
    left: -34px;
    background-image: url(./images/reminder.svg);
    background-repeat: no-repeat;
    background-size: 20px 20px;
  }
}
