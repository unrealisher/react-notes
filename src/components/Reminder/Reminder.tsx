import React from "react";
import moment from "moment";
import "moment/locale/ru";

import styles from "./Reminder.module.scss";

interface IProps {
  reminder: number;
}

const Reminder = (props: IProps): JSX.Element => {
  const { reminder } = props;
  moment.locale("ru");
  return (
    <div className={styles.reminder}>
      <b className={styles.text}>{moment(reminder).fromNow()}</b>
    </div>
  );
};

export default Reminder;
