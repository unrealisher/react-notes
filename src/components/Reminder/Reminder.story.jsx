import React from "react";

import Reminder from "./Reminder";

import { storiesOf } from "@storybook/react";

import "./../../index.scss";

storiesOf("Reminder", module).add("Стандарт", () => (
  <Reminder reminder={1552640400000} />
));
