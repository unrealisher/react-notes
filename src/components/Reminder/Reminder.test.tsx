import React from "react";
import { shallow } from "enzyme";
import { Reminder } from "./Reminder";

describe("Reminder", () => {
  const reminder = 100;
  it("correct render", () => {
    const component = shallow(<Reminder reminder={reminder} />);
    expect(component).toMatchSnapshot();
  });
});
