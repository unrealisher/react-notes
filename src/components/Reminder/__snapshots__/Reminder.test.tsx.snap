// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reminder correct render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Reminder
    reminder={100}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <b
        className="text"
      >
        49 лет назад
      </b>,
      "className": "reminder",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "49 лет назад",
        "className": "text",
      },
      "ref": null,
      "rendered": "49 лет назад",
      "type": "b",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <b
          className="text"
        >
          49 лет назад
        </b>,
        "className": "reminder",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "49 лет назад",
          "className": "text",
        },
        "ref": null,
        "rendered": "49 лет назад",
        "type": "b",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
